base_layer/chat_ffi/src/lib.rs

```diff
@@ -0,0 +1,153 @@
+
+/// Creates a Chat Client
+///
+/// ## Arguments
+/// `config` - The P2PConfig pointer
+/// `identity_file_path` - The path to the node identity file
+/// `db_path` - The path to the db file
+/// `network_str` - The network to connect to
+/// ## Returns
+/// `*mut ChatClient` - Returns a pointer to a ChatClient, note that it returns ptr::null_mut()
+/// if config is null, an error was encountered or if the runtime could not be created
+///
+/// # Safety
+/// The ```client_destroy``` method must be called when finished with a TariWallet to prevent a memory leak
+#[no_mangle]
+pub unsafe extern "C" fn create_chat_client(
+    config: *mut P2pConfig,
+    identity_file_path: *const c_char,
+    db_path: *const c_char,
+    seed_peers: *mut *mut Peer,
+    network_str: *const c_char,
+) -> *mut ClientFFI {
+    let identity_path = PathBuf::from(
+        CStr::from_ptr(identity_file_path)
+            .to_str()
+            .expect("A non-null identity path should be able to convert to a string"),
+    );
+    let mut buf = Vec::new();
+    File::open(identity_path)
+        .expect("Can't open the identity file")
+        .read_to_end(&mut buf)
+        .expect("Can't read the identity file into buffer");
+    let identity: NodeIdentity = serde_json::from_slice(&buf).expect("Can't parse identity file as json");
+
+    let network = Network::from_str(
+        CStr::from_ptr(network_str)
+            .to_str()
+            .expect("A non-null network should be able to be converted to string"),
+    )
+    .expect("Network is invalid");
+
+    let db_path = PathBuf::from(
+        CStr::from_ptr(db_path)
+            .to_str()
+            .expect("A non-null db path should be able to convert to a string"),
+    );
+
+    let mut seed_peers_vec = Vec::new();
+
+    let mut i = 0;
+    while !(*seed_peers.offset(i)).is_null() {
+        let peer = (**seed_peers.offset(i)).clone();
+        seed_peers_vec.push(peer);
+        i += 1;
+    }
+
+    let runtime = Runtime::new().unwrap();
+    let mut client = Client::new(identity, (*config).clone(), seed_peers_vec, db_path, network);
+
+    runtime.block_on(client.initialize());
+
+    let shutdown = Shutdown::new();
+
+    let client_ffi = ClientFFI {
+        client,
+        runtime,
+        shutdown,
+    };
+
+    Box::into_raw(Box::new(client_ffi))
+}
+
+/// Sends a message over a client
+///
+/// ## Arguments
+/// `client` - The Client pointer
+/// `receiver` - A string containing a tari address
+/// `message` - The peer seeds config for the node
+/// ## Returns
+/// `*mut ChatClient` - Returns a pointer to a ChatClient, note that it returns ptr::null_mut()
+/// if config is null, an error was encountered or if the runtime could not be created
+///
+/// # Safety
+/// The ```client_destroy``` method must be called when finished with a TariWallet to prevent a memory leak
+#[no_mangle]
+pub unsafe extern "C" fn send_message(
+    client: &mut ClientFFI,
+    receiver_c_char: *const c_char,
```

Comment:
```suggestion
    receiver_c_char: *TariAdresses,
```
Rather make it take a tari address here, then we can be sure it should work here and we don't have to expect it. We can effectively copy it over from the wallet ffi. It takes tari address ppointers.